server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -292,28 +292,29 @@ private void deleteConfirmed(GarbageCollectionEnvironment gce,
 
   public void collect(GarbageCollectionEnvironment gce) throws TableNotFoundException, IOException {
 
-    String lastCandidate = "";
+    Iterator<String> candidatesIter = getCandidates(gce);
 
-    boolean outOfMemory = true;
-    while (outOfMemory) {
-      List<String> candidates = new ArrayList<>();
-
-      outOfMemory = getCandidates(gce, lastCandidate, candidates);
+    while (candidatesIter.hasNext()) {
+      List<String> batchOfCandidates = gce.readCandidatesThatFitInMemory(candidatesIter);
```

Comment:
This line is what the `try (TraceScope candidatesSpan = Trace.startSpan("getCandidates")) {` should wrap.

Reply:
@ctubbsii, I created a separate PR to address this as the issue was already closed and merged. See #2216.

I'm hoping the IT timeout was a fluke. Not sure if it is included in the Sunny tests, but I have run them several times with no timeout.